### PR TITLE
catkin plugin: nicely handle an invalid rosdistro.

### PR DESCRIPTION
**Pull request by kyrofa**
This PR fixes LP: [#1629957](https://bugs.launchpad.net/snapcraft/+bug/1629957) by verifying that the rosdistro is valid when creating the plugin, and providing a helpful error message if not. Without this PR, if one requests an invalid rosdistro, snapcraft dies with an unfriendly key error.

### CODE REVIEW

File: snapcraft/internal/common.py [hunk 1]
```diff
@@ -268,3 +268,26 @@ def format_path_variable(envvar, paths, prepend, separator):
     return '{envvar}="${envvar}{separator}{paths}"'.format(
         envvar=envvar, separator=separator, paths=combine_paths(
             paths, prepend, separator))
+
+
+def humanize_list(items, conjunction):
```
Comment (elopio) [inline snapcraft/internal/common.py]: could we have a file like formatting.py, or something? common is always. wrong.

**Comment by elopio** [comment]
:+1:, with one comment to maybe find a place that's better than common.

**Comment by sergiusens** [comment]
El 04/10/16 a las 16:12, Leo Arias escribió:
>
> *@elopio* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In snapcraft/internal/common.py 
> <https://github.com/snapcore/snapcraft/pull/851#pullrequestreview-2782857>:
>
> > @@ -268,3 +268,26 @@ def format_path_variable(envvar, paths, prepend, separator):
>       return '{envvar}="${envvar}{separator}{paths}"'.format(
>           envvar=envvar, separator=separator, paths=combine_paths(
>               paths, prepend, separator))
> +
> +
> +def humanize_list(items, conjunction):
>
> could we have a file like formatting.py, or something? common is 
> always. wrong.
>

+1 on this.

File: snapcraft/internal/common.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/internal/common.py]: Yeah, good call. Doing now.

Comment (kyrofa) [inline snapcraft/internal/common.py]: Done.

File: snapcraft/formatting_utils.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    :param str prepend: String to prepend to each path in the string.
+    :param str separator: String to place between each path in the string.
+    """
+
+    paths = ['{}{}'.format(prepend, p) for p in paths]
+    return separator.join(paths)
+
+
+def format_path_variable(envvar, paths, prepend, separator):
+    """Return an path-like environment variable definition that appends.
```
Comment (elopio) [inline snapcraft/formatting_utils.py]: Typo: s/an/a

**Pull request merged**